File: python/Ice/inheritance/server/m_node.py
# Copyright (c) ZeroC, Inc.

import Ice
from Filesystem import Node


class MNode(Node):
    """
    Provides an in-memory implementation of the Slice interface Node.
    """

    def __init__(self, name: str):
        """
        Initializes an MNode with the given name.

        Parameters
        ----------
        name : str
            The name of the file.
        """
        self._name = name

    def name(self, current: Ice.Current) -> str:
        return self._name

File: csharp/Ice/Inheritance/Server/MNode.cs
// Copyright (c) ZeroC, Inc.

namespace Server;

/// <summary>Provides an in-memory implementation of the Slice interface Node.</summary>
internal class MNode : Filesystem.NodeDisp_
{
    private readonly string _name;

    /// <inheritdoc/>
    public override string Name(Ice.Current current) => _name;

    /// <summary>Constructs an MNode.</summary>
    /// <param name="name">The name of this node.</param>
    internal MNode(string name) => _name = name;
}
